smart_open/tests/test_smart_open.py

```diff
@@ -259,7 +259,34 @@ def test_http_bz2(self):
                       body=compressed_data, stream=True)
         smart_open_object = smart_open.smart_open("http://127.0.0.1/data.bz2")
 
-        # decompress the gzip and get the same md5 hash
+        # decompress the bzip2 and get the same md5 hash
+        self.assertEqual(smart_open_object.read(), test_string)
+
+    @responses.activate
+    def test_http_xz(self):
+        """Can open xz via http?"""
+        test_string = b'Hello World Compressed.'
+        #
+        # TODO: why are these tests writing to temporary files?  We can do the
+        # lzma compression in memory.
+        #
+        with tempfile.NamedTemporaryFile('wb', suffix='.xz', delete=False) as infile:
+            test_file = infile.name
+
+        with smart_open.smart_open(test_file, 'wb') as outfile:
+            outfile.write(test_string)
+
+        with open(test_file, 'rb') as infile:
+            compressed_data = infile.read()
+
+        if os.path.isfile(test_file):
+            os.unlink(test_file)
+
+        responses.add(responses.GET, "http://127.0.0.1/data.xz",
```

Comment:
No need to line break here. We have a line limit of 120 characters.

Reply:
Again, copy-pasted from `test_http_bz`. Shall I edit both functions at once or submit a separate PR?